# This patch locks the screen whenever the system suspends or hybernates.
diff -ur ./lxqtpower/lxqtpower.cpp ../liblxqt-patched/lxqtpower/lxqtpower.cpp
--- ./lxqtpower/lxqtpower.cpp	2015-07-29 15:44:01.025128882 +0200
+++ ../liblxqt-patched/lxqtpower/lxqtpower.cpp	2015-07-29 15:54:21.725369061 +0200
@@ -34,8 +34,10 @@
 using namespace LxQt;
 
 Power::Power(QObject *parent) :
-    QObject(parent)
+    QObject(parent),
+    mScreenSaver(this)
 {
+    connect(&mScreenSaver, SIGNAL(done()), &mLoop, SLOT(quit()));
     mProviders.append(new CustomProvider(this));
     mProviders.append(new UPowerProvider(this));
     mProviders.append(new ConsoleKitProvider(this));
@@ -66,11 +68,18 @@
 {
     foreach(PowerProvider* provider, mProviders)
     {
-        if (provider->canAction(action) &&
-            provider->doAction(action)
-           )
+        if (provider->canAction(action))
         {
-            return true;
+            if (action == PowerSuspend || action == PowerHibernate)
+            {
+                mScreenSaver.lockScreen();
+                mLoop.exec();
+            }
+            if (provider->doAction(action))
+            {
+                return true;
+            }
+            return false;
         }
     }
     return false;
diff -ur ./lxqtpower/lxqtpower.h ../liblxqt-patched/lxqtpower/lxqtpower.h
--- ./lxqtpower/lxqtpower.h	2015-07-29 15:43:45.360324719 +0200
+++ ../liblxqt-patched/lxqtpower/lxqtpower.h	2015-07-29 15:53:55.476697215 +0200
@@ -31,6 +31,8 @@
 
 #include <QObject>
 #include <QList>
+#include <QEventLoop>
+#include "lxqtscreensaver.h"
 #include "lxqtglobals.h"
 
 namespace LxQt
@@ -101,6 +103,8 @@
 
 private:
     QList<PowerProvider*> mProviders;
+    LxQt::ScreenSaver mScreenSaver;
+    QEventLoop mLoop;
 };
 
 } // namespace LxQt
